imports/plugins/included/surcharges/server/no-meteor/util/getSurchargesTotalForCart.js

```diff
@@ -0,0 +1,26 @@
+import ReactionError from "@reactioncommerce/reaction-error";
+
+/**
+ * @summary Calculates total surcharge amount for a cart based on all surcharges
+ *   that have been applied to it
+ * @param {Object} context Context object
+ * @param {String} cartId Cart ID
+ * @returns {Object} Object with `surcharges` array and `total`
+ */
+export default async function getSurchargesTotalForCart(context, cartId) {
```

Comment:
So if no cart is available, should I throw an error similar to [how discounts is doing it](https://github.com/reactioncommerce/reaction/blob/feat-kieckhafer-surcharges/imports/plugins/core/discounts/server/no-meteor/util/getDiscountsTotalForCart.js#L14-L17)?

I'm not following how we would attach surcharges if no cart is available.

Reply:
@kieckhafer Don't look at the way discounts is done because that was a hack to keep it working for now.

You should think of an order as its own thing that you calculate surcharges for. Actually the primary thing we calculate surcharges for. It helps to remember that when "orders" is someday its own microservice, it probably won't even be aware of what a cart is. If your system does not have the concept of a cart, you could remove cart plugin and still place orders.

In practical terms, it would be similar to `addShipmentMethodToGroup` and `addTaxesToGroup` functions in `placeOrder`. Add a new similar function, and in it you will convert order to commonOrder, and then call `await getSurcharges(context, { commonOrder })`. Tweak the existing `getSurcharges` function to take `commonOrder` rather than `cart`, making it useable for either cart or order. You can convert cart to commonOrder in the afterCartUpdate function.

Also, though, `placeOrder.js` should not import anything that's in `included` because the plugin might be removed. So you can instead have the surcharges plugin register a function of type "getSurcharges" and use `getFunctionsOfType` to run all such functions and merge the resulting arrays.

Reply:
Should this be done inside the map of each `fulfillmentGroup`, or should it be outside of this since the surcharge applies to the order?

https://github.com/reactioncommerce/reaction/blob/feat-kieckhafer-surcharges/imports/plugins/core/orders/server/no-meteor/mutations/placeOrder.js#L466

If outside, what would we pass to the `xformOrderGroupToCommonOrder` function as the `group`?